src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/HeaderNode.cs

```diff
@@ -94,7 +150,23 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                 _items.MergeSort((x, y) => Comparer<int>.Default.Compare((int)x.Id, (int)y.Id));
 
             // ReadyToRunHeader.Flags
-            builder.EmitInt((int)_flags);
+            int flagsInt = (int)_flags;
+            if (!relocsOnly)
+            {
+                if (_shouldAddSkipTypeValidationFlag.Result.canSkipValidation)
+                {
+                    flagsInt |= (int)ReadyToRunFlags.READYTORUN_FLAG_SkipTypeValidation;
+                }
+                else
+                {
+                    if (factory.OptimizationFlags.TypeValidation == TypeValidationRule.AutomaticWithLogging)
+                    {
+                        foreach (string reason in _shouldAddSkipTypeValidationFlag.Result.reasons)
```

Comment:
I might be missing part of the intended logic here - according to the naming I would expect <code>_shouldAddSkipTypeValidationFlag</code> to represent some form of vetting whether we can drop the type validation and demonstrate the rationale if we do; why do we dump the reasons in the <code>else</code> block instead?

Reply:
Added a comment. These are logging reasons why we aren't able to skip validation. Logging reasons we could skip would be really boring. (It would just be a list of types that are considered ok, which isn't all that useful for anybody)